src/test/regress/sql/foreign_key_to_reference_table.sql

```diff
@@ -467,6 +565,23 @@ DROP TABLE referenced_table CASCADE;
 DROP TABLE referencing_table CASCADE;
 DROP TABLE referencing_table2 CASCADE;
 
+-- Check if the same fkeys are created with create_distributed_table
+CREATE TABLE referenced_table(test_column int, test_column2 int UNIQUE, PRIMARY KEY(test_column));
+CREATE TABLE referencing_table(id int PRIMARY KEY, ref_id int, FOREIGN KEY (id) REFERENCES referenced_table(test_column) ON DELETE CASCADE);
+CREATE TABLE referencing_table2(id int, ref_id int, FOREIGN KEY (ref_id) REFERENCES referenced_table(test_column2) ON DELETE CASCADE, FOREIGN KEY (id) REFERENCES referencing_table(id) ON DELETE CASCADE);
+SELECT create_reference_table('referenced_table');
+BEGIN;
+  SET LOCAL citus.multi_shard_modify_mode TO 'sequential';
```

Comment:
why do we set the mode to sequential? I think it's OK to have a single test with `sequential ` mode, but just manually ensure that this test passes without explicit set as well.